### ISSUE TITLE
Can't edit MySQL virtual columns with certain characters in expression - incorrect escaping

### ISSUE THREAD

**Issue by lemon-juice** [dbeaver/dbeaver#21648]
### Description

Let's say we have a table like this:

```sql
CREATE TABLE `testtable` (
  `id` mediumint unsigned NOT NULL AUTO_INCREMENT,
  `doc_nr` varchar(20) COLLATE utf8mb4_unicode_ci NOT NULL,
  `doc_counter` smallint unsigned GENERATED ALWAYS AS (cast(regexp_substr(`doc_nr`,_utf8mb4'\\d+') as unsigned)) VIRTUAL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```

The `doc_counter` column extracts a number from the first digits found in `doc_nr`. But it looks like certain characters used in the expression like ' and \ make DBeaver escape too much. If I try to edit the virtual column DBeaver creates DDL with bad syntax. For example, let's try to rename the column and DBeaver creates this SQL:

```SQL
ALTER TABLE `20230209`.testtable CHANGE doc_counter doc_counter2 smallint unsigned GENERATED ALWAYS AS (cast(regexp_substr(`doc_nr`,_utf8mb4\'\\\\d+\') as unsigned)) VIRTUAL NULL;
```
This will not execute because of syntax error. There are unnecessary backslashes in the regex string. The correct query is like this:
```SQL
ALTER TABLE `20230209`.testtable CHANGE doc_counter doc_counter2 smallint unsigned GENERATED ALWAYS AS (cast(regexp_substr(`doc_nr`,_utf8mb4'\\d+') as unsigned)) VIRTUAL NULL;
```
The incorrect over-escaped expression also shows in the table column properties:
```
cast(regexp_substr(`doc_nr`,_utf8mb4\'\\\\d+\') as unsigned)
```
As a result it is not possible to edit such columns in DBeaver without manually fixing the generated query.

### DBeaver Version

Community Edition 23.2.3

### Operating System

Windows 10

### Database and driver

MySQL 8.0.32

**Comment by LonwoLonwo** [comment]
Hello @lemon-juice 

Thanks for the bug report.

DBeaver reads this expression from the information_schema.COLUMNS, and, as I can see, it returns an incorrect value.

![2023-10-25 12_03_54-DBeaver Ultimate 23 2 3 - _Google Cloud SQL - MySQL - test-mysql-instance_ Scrip](https://github.com/dbeaver/dbeaver/assets/45152336/804fe1a3-5d42-4055-8904-3f74d569e437)

We can try to fix it somehow, probably.

Also, you can add your report here: https://bugs.mysql.com/

**Comment by lemon-juice** [comment]
Indeed, I've found it's a known regression bug in MySQL 8: https://bugs.mysql.com/bug.php?id=100607. So it's an unfortunate state of affairs because we do not know when MySQL is going to fix it and looking at the past it may take years...

So what can be done? I think the best solution for us users would be if DBeaver checked the MySQL version and applied unescape on the expression if the version is 8.x. I know it's the kind of fix that is undesirable because it doesn't really belong to DBeaver and it needs tracking at what version it will get actually fixed by the database vendor but what else can be done? Even if MySQL fixed this bug very soon (very unlikely) then there are already a large number of MySQL 8.x installs affected.

**Comment by LonwoLonwo** [comment]
For now, unfortunately, I can not find any good options to fix it easily. I do not see this column meta info in other system tables/views.
So, or we need to parse the table DDL, or we need to remove somehow extra escaping from an expression. 
Both options look bad.

**Comment by lemon-juice** [comment]
You are right, there are no good options without hacks. But I think they should not be very difficult. The logic could be like this:

```
IF MySQL_version == 8.x THEN replace chars:
  \' => '
  \\ => \
```
Or the other hack with parsing DDL:

1. Read the DDL until line starts with \`column_name\`.
2. Get text between the first ` AS ` and final `VIRTUAL` or `STORED`.
3. Strip parentheses.
